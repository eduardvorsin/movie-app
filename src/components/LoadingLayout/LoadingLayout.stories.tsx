import type { Meta, StoryObj } from '@storybook/react';
import LoadingLayout from './LoadingLayout';

const meta: Meta<typeof LoadingLayout> = {
	title: 'components/LoadingLayout',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for the LoadingLayout'
		},
		testId: {
			description: 'Id for testing the component',
		},
	},
	component: LoadingLayout,
	parameters: {
		docs: {
			description: {
				component: 'The component responsible for displaying the page loading status. It is shown both when the page is initially loaded and when navigating through pages',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof LoadingLayout>;

export const Default: Story = {
	args: {
		className: '',
		testId: '',
	},
};