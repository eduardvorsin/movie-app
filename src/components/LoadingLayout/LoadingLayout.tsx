import Spinner from '@/components/Spinner/Spinner';

export default function LoadingLayout() {
	return (
		<div className='w-full min-h-[100vh] relative'>
			<Spinner
				className='absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2'
				size='xlarge'
			/>
		</div>
	);
}