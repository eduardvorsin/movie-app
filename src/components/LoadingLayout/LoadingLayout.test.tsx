import { render, screen } from "@testing-library/react";
import LoadingLayout from "./LoadingLayout";

describe('LoadingLayout tests', () => {
	it('is rendered correctly', () => {
		render(<LoadingLayout testId='test-loading-layout' />);

		expect(screen.getByTestId<HTMLDivElement>('test-loading-layout')).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		render(<LoadingLayout testId='test-loading-layout' />);

		expect(screen.getByTestId<HTMLDivElement>('test-loading-layout')).toMatchSnapshot();
	});
});