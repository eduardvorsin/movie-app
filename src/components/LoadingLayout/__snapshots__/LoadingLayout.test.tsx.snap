// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingLayout tests is a basic snapshot 1`] = `
<div
  class="w-full min-h-[100vh] relative undefined"
  data-testid="test-loading-layout"
>
  <span
    class="inline-flex align-middle dark:text-blue-300 text-blue-800 transition-colors duration-150 w-24 h-24 absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2"
  >
    <svg
      class="animate-spin stroke-[0.75] fill-none stroke-current"
      viewBox="0 0 16 16"
    >
      <circle
        cx="8"
        cy="8"
        r="7"
        stroke-dasharray="60"
        stroke-dashoffset="50"
        stroke-linecap="round"
      />
    </svg>
  </span>
</div>
`;
